libmscore/measure.cpp

```diff
@@ -578,7 +578,7 @@ void Measure::layout2()
                         smn = system()->firstMeasure() == this;
                   else {
                         smn = (_no == 0 && score()->styleB(StyleIdx::showMeasureNumberOne)) ||
-                              ( ((_no+1) % score()->style(StyleIdx::measureNumberInterval).toInt()) == 0 );
+                              ( ((_no+1) % score()->style(StyleIdx::measureNumberInterval).toInt()) == score()->styleB(StyleIdx::showMeasureNumberOne)?1:0 );
```

Comment:
Maybe add white spaces around 1 and 0 or extract the == statement to separate variable? If no, at least adding brackets around the == statement will solve possible misunderstanding in calculating sequence.

Reply:
Done, white space plus brackets